cmd/ulimit_unix.go

```diff
@@ -15,37 +15,87 @@ func init() {
 	fileDescriptorF = checkAndSetUlimitUnix
 }
 
+// In this entire file, by 'ulimit' we mean the "num open files ulimit" (RLIMIT_NOFILE)
 func checkAndSetUlimitUnix() {
 	logger := zerolog.Logger
-	if err := setUlimit(ulimitUnix); err != nil {
-		logger.Err(fmt.Errorf("error setting ulimit: %w", err))
+
+	rlimit, err := getUlimit()
+	if err != nil {
+		logger.Err(err).Msg("failed getting ulimit")
+		return
+	}
+	logger.Info().Uint64("hard_ulimit", rlimit.Max).Uint64("soft_ulimit", rlimit.Cur).Msg("limits (before adjustment)")
+
+	if err := setUlimit(); err != nil {
+		logger.Err(err).Msg("failed setting ulimit")
+	}
+
+	rlimit, err = getUlimit()
+	if err != nil {
+		logger.Err(err).Msg("failed getting ulimit")
+		return
+	}
+	logger.Info().Uint64("hard_ulimit", rlimit.Max).Uint64("soft_ulimit", rlimit.Cur).Msg("limits (after adjustment)")
+}
+
+func setUlimit() error {
+	logger := zerolog.Logger
+
+	// Setting the hard ulimit is very likely to fail (usually requires root or even more arcane permissions).
+	// If it fails we just log at "INFO" level.
+	if err := setHardUlimit(); err != nil {
+		// Just an info log here, since it's really not actionable.
+		logger.Info().AnErr("err", err).Msg("failed setting hard ulimit")
 	}
+
+	return setSoftUlimit()
 }
 
-func setUlimit(ulimit uint64) error {
+func setHardUlimit() error {
 	logger := zerolog.Logger
-	rLimit, err := GetUlimit()
+	rLimit, err := getUlimit()
```

Comment:
[non-blocking] Can we avoid this extra call to `getUlimit`? We just read it in `checkAndSetUlimitUnix`.
We could pass it via an argument

Reply:
Leaving as-is for now, but will add the logging.
[Only the syscall in setHardLimit can be avoided - the syscall in setSoftLimit needs to happen because it has no way of knowing if setHardLimit succeeded or not].
[Because only one of them can be avoided, I'd prefer to have a cleaner interface and consistency in code and behaviour between them]...